.github/workflows/main.yml

```diff
@@ -14,21 +14,37 @@ jobs:
     name: build without warnings
     runs-on: ubuntu-latest
     steps:
+    - uses: actions/cache@v2
+      with:
+        path: ~/.nuget/packages
+        key: ${{ runner.os }}-nuget
     - uses: actions/checkout@v2
     - name: Setup .NET Core
       uses: actions/setup-dotnet@v1
       with:
         dotnet-version: ${{ env.DOTNET_VERSION }}
 
     - name: Install dependencies
-      run: dotnet restore src/Framework/Framework/DotVVM.Framework.csproj
+      run: dotnet restore ci/linux/Linux.sln
       
     - name: Build (without warnings)
-      run: dotnet build src/Framework/Framework/DotVVM.Framework.csproj --configuration Release --no-restore --no-incremental /property:WarningLevel=0
+      run: dotnet build src/Framework/Framework --configuration Release --no-restore --no-incremental /property:WarningLevel=0
     - name: Build
-      run: dotnet build src/Framework/Framework/DotVVM.Framework.csproj --configuration Release --no-restore --no-incremental --framework netstandard2.1 /WarnAsError
+      run: dotnet build src/Framework/Framework --configuration Release --no-restore --no-incremental /WarnAsError
     - name: Build (Debug)
-      run: dotnet build src/Framework/Framework/DotVVM.Framework.csproj --configuration Debug --no-restore --no-incremental --framework netstandard2.1 /WarnAsError
+      run: dotnet build src/Framework/Framework --configuration Debug --no-restore --no-incremental /WarnAsError
+    - name: Build Hosting.AspNetCore
+      run: dotnet build src/Framework/Hosting.AspNetCore --no-incremental /WarnAsError
+    - name: Build Testing
+      run: dotnet build src/Framework/Testing --no-incremental /WarnAsError
+    - name: Build Tracing.MiniProfiler
+      run: dotnet build src/Tracing/MiniProfiler.AspNetCore --no-incremental /WarnAsError
+    - name: Build Tracing.ApplicationInsights
+      run: dotnet build src/Tracing/ApplicationInsights.AspNetCore --no-incremental /WarnAsError
+    - name: Build Dynamic Data
+      run: dotnet build src/DynamicData/DynamicData --no-incremental /WarnAsError
+    - name: Build Compiler
+      run: dotnet build src/Tools/Compiler --no-incremental /WarnAsError
```

Comment:
Wouldn't it be simple to just `dotnet build ci/linux/Linux.sln --no-incremental /WarnAsError`? (Feel free to add whatever project you want to that solution.)

Reply:
This way it shows which project causes the problem in the UI. And I'd have to remove some projects, because I don't want to strictly check warning in samples, tests, ...

Reply:
Fair enough. I see no issues with merging this then. :)